scripts/compare_benchmarks.sh

```
@@ -94,16 +94,55 @@ echo "Using benchmark results file: $summary_file"
+fi
+
+echo "Using baseline file: $BASELINE_FILE"
+
+# Parse baseline values from the file
+# Function to extract time value from baseline (converts to nanoseconds)
+extract_baseline_time() {
+    local points="$1"
+    local dimension="$2"
+    
+    # Look for the section header and extract the mean time value
+    local time_line=$(grep -A 1 "=== $points Points (${dimension}D) ===" "$BASELINE_FILE" | grep "Time:" | head -1)
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)